include/cuda/std/detail/libcxx/include/support/atomic/atomic_cuda_local.h

```diff
@@ -0,0 +1,140 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+#ifndef __LIBCUDACXX_ATOMIC_CUDA_LOCAL_H
+#define __LIBCUDACXX_ATOMIC_CUDA_LOCAL_H
+
+__device__ static inline int __memcmp(const void* __lhs, const void* __rhs, size_t __count)
```

Comment:
This could/should use `__cuda_memcmp`. I'll leave it up to you to decide which implementation is better. 

https://github.com/NVIDIA/libcudacxx/blob/a13db9aeab643e01498e5fcc3f454bbb9ea1626c/include/cuda/std/detail/libcxx/include/support/atomic/atomic_cuda.h#L401-L419

Reply:
They also suggested that they would prefer to expose memcmp through libcu++ instead of as an intrinsic, given that with that change it can be implemented reliably on the library side, so it might make sense to expose that through `cuda::std::memcmp` (if it isn't already).